modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java

```diff
@@ -256,6 +320,71 @@ public void testGetMetaOutboundConnector() throws IOException {
+        body = body.replace(FEDERATED_AUTHENTICATOR_PLACEHOLDER,
+                userDefinedAuthenticatorPayload.convertToJasonPayload());
+        body = body.replace(IDP_NAME_PLACEHOLDER, IDP_NAME);
+        Response response = getResponseOfPost(IDP_API_BASE_PATH, body);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_CREATED)
+                .header(HttpHeaders.LOCATION, notNullValue());
+
+        String location = response.getHeader(HttpHeaders.LOCATION);
+        assertNotNull(location);
+        customIdPId = location.substring(location.lastIndexOf("/") + 1);
+        assertNotNull(customIdPId);
+    }
+
+    @Test(dependsOnMethods = "testAddIdPWithUserDefinedAuthenticator")
+    public void testGetUserDefinedAuthenticatorsOfIdP() {
+
+        Response response = getResponseOfGet(IDP_API_BASE_PATH + PATH_SEPARATOR + customIdPId +
+                PATH_SEPARATOR + IDP_FEDERATED_AUTHENTICATORS_PATH);
+
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .body("defaultAuthenticatorId", equalTo(FEDERATED_AUTHENTICATOR_ID))
+                .body("authenticators.find { it.authenticatorId == '" + FEDERATED_AUTHENTICATOR_ID + "' }.name",
+                        equalTo(new String(Base64.getDecoder().decode(FEDERATED_AUTHENTICATOR_ID))))
+                .body("authenticators.find { it.authenticatorId == '" + FEDERATED_AUTHENTICATOR_ID + "' }.isEnabled",
+                        equalTo(true));
+    }
+
+    @Test(dependsOnMethods = "testGetUserDefinedAuthenticatorsOfIdP")
+    public void testUpdateUserDefinedAuthenticatorOfIdP() throws JsonProcessingException {
+
+        Response response = getResponseOfPut(IDP_API_BASE_PATH + PATH_SEPARATOR + customIdPId +
+                        PATH_SEPARATOR + IDP_FEDERATED_AUTHENTICATORS_PATH + PATH_SEPARATOR + FEDERATED_AUTHENTICATOR_ID,
+                createUserDefinedAuthenticatorPayload(UPDATED_ENDPOINT_URI).convertToJasonPayload());
+
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .body("authenticatorId", equalTo(FEDERATED_AUTHENTICATOR_ID))
+                .body("name", equalTo(new String(Base64.getDecoder().decode(FEDERATED_AUTHENTICATOR_ID))))
+                .body("endpoint.uri", equalTo(UPDATED_ENDPOINT_URI));
+    }
+
+    @Test(dependsOnMethods = "testUpdateUserDefinedAuthenticatorOfIdP")
+    public void testDeleteIdPWithUserDefinedAuthenticator() {
+
+        Response response = getResponseOfDelete(IDP_API_BASE_PATH + PATH_SEPARATOR + customIdPId);
+
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_NO_CONTENT);
+    }
+
```

Comment:
As we discussed offline add a delete test case as well. and invoke self ref of fed authenticator and make sure, it is delete properly.

Reply:
This is fixed with 38608298188e4987329c170552b4a0de9fb72daf